File: CivitasJava/civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;

public class SorpresaPagarCobrar extends Sorpresa {
    private int valor;
    
    SorpresaPagarCobrar(int val, String tex){
        super(tex);
        valor = val;
    }
    
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) throws Exception{
        if ( jugadorCorrecto(actual, todos) ){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
}

File: CivitasRuby/sorpresa_pagar_cobrar.rb
require_relative 'sorpresa'

module Civitas
class SorpresaPagarCobrar < Sorpresa
  
  def initialize(valor, texto)
    super(texto)
    @valor = valor
  end
  
  def aplicar_a_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      informe(actual, todos)
      todos[actual].modificar_saldo(@valor)
    end
  end
  
  public_class_method :new
end
end
